cpp/ql/src/semmle/code/cpp/dataflow/internal/FlowVar.qll

```diff
@@ -249,6 +249,9 @@ module FlowVar_internal {
           result = descendentDef.getAUse(v)
         )
       )
+      or
+      parameterUsedInConstructorFieldInit(v, result) and
+      def.definedByParameter(v)
```

Comment:
Are `Parameter`s special here?  I mean are things going to work if a `ConstructorFieldInit` accesses a member or local variable?

Reply:
`Parameter`s and `this` are special because they are the only variables that can receive a value before before the member initializers lists run. **Local variables** can't be declared until the body, which comes after the member initializers. **Member variables** come from `this` or from parameters, and `this` already receives special treatment (`TInstanceParameterNode`).